libnd4j/include/ops/declarable/helpers/cuda/random.cu

```diff
@@ -181,6 +181,56 @@ namespace helpers {
     }
 
     BUILD_SINGLE_TEMPLATE(template void fillRandomPoisson_, (LaunchContext* context, graph::RandomGenerator& rng, NDArray* lambda, NDArray* output), FLOAT_NATIVE);
+
+    template <typename T>
+    static __global__ void fillUniformKernel(graph::RandomGenerator* devRng, T from, T to, T* output, Nd4jLong* outputShape) {
+        auto start = blockIdx.x * blockDim.x + threadIdx.x;
+        auto step = blockDim.x * gridDim.x;
+
+        __shared__ Nd4jLong outputLen;
+
+        if (0 == threadIdx.x) {
+            outputLen = shape::length(outputShape);
+        }
+        __syncthreads();
+
+        for (auto i = start; i < outputLen; i += step) {
+            auto zIndex = shape::getIndexOffset(i, outputShape);
+            output[zIndex] = devRng->relativeT<T>(i, from, to);
+        }
+
+    }
+
+    template <typename T>
+    static void fillRandomUniform_(LaunchContext* context, graph::RandomGenerator& rng, NDArray* min, NDArray* max, NDArray* output) {
+        T minVal = T(0);
+        T maxVal = DataTypeUtils::infOrMax<T>();
+        if (min)
+            minVal = min->t<T>(0);
+        if (max)
+            maxVal = max->t<T>(0);
+
+        if (output->isR())
+            RandomLauncher::fillUniform(context, rng, output, minVal, maxVal);
+        else {
+            auto stream = context->getCudaStream();
+            graph::RandomGenerator *devRng;
+            cudaMalloc(&devRng, sizeof(graph::RandomGenerator));
+            cudaMemcpy(devRng, &rng, sizeof(graph::RandomGenerator), cudaMemcpyHostToDevice);
+            auto outputBuf = output->dataBuffer()->specialAsT<T>();
+            auto outputShape = output->specialShapeInfo();
+            fillUniformKernel<T><<<128, 128, 128, *stream>>>(devRng, minVal, maxVal, outputBuf, outputShape);
+            cudaFree(devRng);
```

Comment:
synchronize before cudaFree call

Reply:
Added error checking and synchronize stuff.